packages/core/contracts/cross-chain-oracle/chain-adapters/Optimism_ChildMessenger.sol

```diff
@@ -53,7 +54,11 @@ contract Optimism_ChildMessenger is OVM_CrossDomainEnabled, ChildMessengerInterf
      * @dev The caller of this function must be the parent messenger, over the canonical bridge.
      * @param newParentMessenger address of the new parent messenger, deployed on L1.
      */
-    function setParentMessenger(address newParentMessenger) public onlyFromCrossDomainAccount(parentMessenger) {
+    function setParentMessenger(address newParentMessenger)
+        public
+        onlyFromCrossDomainAccount(parentMessenger)
+        nonReentrant()
```

Comment:
I think these `nonReentrant` modifies are redundant (and potentially dangerous) for us to user here. these methods are access controlled via the `onlyFromCrossDomainAccount` modifier so why do we need this entry guard?

Reply:
I don't see that they're dangerous, perhaps they're overkill, but why would reentrancy ever happen?